### ISSUE TITLE
Assertion failed 

### ISSUE THREAD

**Issue by emmannaemeka** [bcgsc/abyss#470]
# Please report
Hello, trying to run abyss and encountered this error. How can I solve it?

abyss-pe k=96 name=cgf9 B=2G in='f9_R1.fastq    f9_r2.fastq'
abyss-rresolver-short  -b2G -f0.8 -j2 -k96 \
	 -h cgf9-1-rr --dot -c cgf9-1-rr.fa \
	-g cgf9-1-rr.dot cgf9-1.fa cgf9-1.dot f9_R1.fastq f9_r2.fastq 
Assertion failed: (num_vertices(g) > 0), function read_dot, file DotIO.h, line 307.
make: *** [cgf9-1-rr.dot] Abort trap: 6

- [ ] version of ABySS with `abyss-pe version`
- [ ] distribution of Linux with `lsb_release -d`

# Assembly error

- [ ] complete `abyss-pe` command line
- [ ] last 20 lines of the output of `abyss-pe`
- [ ] number of sequenced bases
- [ ] estimated genome size and ploidy
- [ ] estimated sequencing depth of coverage

# Build error

Consider installing ABySS using [Homebrew](https://brew.sh) on either Linux or macOS with `brew install abyss`, or using [Bioconda](https://bioconda.github.io) with `conda install abyss`.

- [ ] Have you tried installing ABySS using Brew or Bioconda?
- [ ] version of GCC or compiler with `gcc --version`
- [ ] complete `./configure` command line
- [ ] last 20 lines of the output of `./configure`
- [ ] last 20 lines of the output of `make`

**Comment by lcoombe** [comment]
Hi @emmannaemeka,

Could you please report your full log, your ABySS version and how you installed ABySS?

Thank you for your interest in ABySS!
Lauren

**Comment by lcoombe** [comment]
Thanks for your installation info. Are you also able to share the full log and ABySS version?

**Comment by emmannaemeka** [comment]
ABYSS (ABySS) 2.3.7

Written by Jared Simpson, Shaun Jackman and Anthony Raymond.


Copyright 2014 Canada's Michael Smith Genome Sciences Centre

(base) emmanuelnnadi@ bioinformatics_tools %


Nnaemeka Emmanuel Nnadi,Ph.D
Department of Microbiology,
Faculty of Natural and Applied Science,
Plateau State University, Bokkos, Plateau State, Nigeria.
+2348068124819
Publications:
https://scholar.google.com/citations?user=5mzOgPQAAAAJ&hl=en
https://orcid.org/0000-0002-8424-4859

 https://www.researchgate.net/profile/Emmanuel_Nnadi/publications

**Comment by lcoombe** [comment]
Ok thank you, and like I mentioned could you also please share your full log? By that, I mean the full command that you ran, and all standard error/log messages printed by ABySS.

**Comment by emmannaemeka** [comment]
abyss-pe name=cgb9 k=96 B=2G in='b9_R1.fastq.gz b9_r2.fastq.gz'

abyss-stack-size 65536 abyss-bloom-dbg -k96 -q3  -b2G -j2  b9_R1.fastq.gz
b9_r2.fastq.gz  > cgb9-1.fa

/usr/local/bin/abyss-stack-size: line 19: ulimit: stack size: cannot modify
limit: Operation not permitted

Running with max stack size of 65532 KB: abyss-bloom-dbg -k96 -q3 -b2G -j2
b9_R1.fastq.gz b9_r2.fastq.gz

b9_R1.fastq.gz:1: error: Expected either `>' or `@' or 11 fields

and saw `' and 2 fields near

???Ɏ?L????????4???Ԡ


????????A?R*E=o?wJd4;???+V?&??????????????????ǿ????????????????????????????????????????_????????????????????????P???b*???????????%Q??????%s?^??y_.??4?l?C


?????z?4??yi???q?w?]?r???????????_??????kks0????????U???=[:?????????$OAw????/??f/?Ƞz?

                 h??KN????

                          zZ^??C


??T?D??w??Rt)??wh{9??%{?]@??;?.c??r??#6??????o?[D?sb??o?~i$??r?SY??Ş???????ᄁ??????p????(????
;Mk??H

 ***@***.***~Ahpo?15`

make: *** [cgb9-1.fa] Error 1
Nnaemeka Emmanuel Nnadi,Ph.D
Department of Microbiology,
Faculty of Natural and Applied Science,
Plateau State University, Bokkos, Plateau State, Nigeria.
+2348068124819
Publications:
https://scholar.google.com/citations?user=5mzOgPQAAAAJ&hl=en
https://orcid.org/0000-0002-8424-4859

 https://www.researchgate.net/profile/Emmanuel_Nnadi/publications

**Comment by emmannaemeka** [comment]
Again I tried another sequences

abyss-pe name=cgf9 k=96 B=2G in='f9_R1.fastq.gz f9_r2.fastq.gz'

abyss-rresolver-short  -b2G -f0.8 -j2 -k96 \

-h cgf9-1-rr --dot -c cgf9-1-rr.fa \

-g cgf9-1-rr.dot cgf9-1.fa cgf9-1.dot f9_R1.fastq.gz f9_r2.fastq.gz

Assertion failed: (num_vertices(g) > 0), function read_dot, file DotIO.h,
line 307.

make: *** [cgf9-1-rr.dot] Abort trap: 6



Nnaemeka Emmanuel Nnadi,Ph.D
Department of Microbiology,
Faculty of Natural and Applied Science,
Plateau State University, Bokkos, Plateau State, Nigeria.
+2348068124819
Publications:
https://scholar.google.com/citations?user=5mzOgPQAAAAJ&hl=en
https://orcid.org/0000-0002-8424-4859

 https://www.researchgate.net/profile/Emmanuel_Nnadi/publications

**Comment by lcoombe** [comment]
Ok, thanks. So that is an error there in the first stage of ABySS (`abyss-bloom-dbg`). It is indicating an issue with your input file format: 
```
b9_R1.fastq.gz:1: error: Expected either `>' or `@' or 11 fields
```
Can you double check that your input files adhere to the conventional fastq format, and are properly gzipped?

If you get a failure in that early stage of ABySS, it is best to delete all the output files to allow ABySS to properly start from the beginning. I suspect that is what may have also caused the second failure, as I can see that (if that's the full log), ABySS is not starting from the beginning of the assembly process, as expected.

So to summarize:
- Delete all the ABySS output files from any early failures
- There appears to be an issue with the file `b9_R1.fastq.gz`, so take a look at that and ensure that the formatting is proper

**Comment by emmannaemeka** [comment]
I re-ran it and

abyss-pe name=cgf9 k=96 B=2G in='f9_R1.fastq.gz f9_r2.fastq.gz'

abyss-stack-size 65536 abyss-bloom-dbg -k96 -q3  -b2G -j2  f9_R1.fastq.gz
f9_r2.fastq.gz  > cgf9-1.fa

/usr/local/bin/abyss-stack-size: line 19: ulimit: stack size: cannot modify
limit: Operation not permitted

Running with max stack size of 65532 KB: abyss-bloom-dbg -k96 -q3 -b2G -j2
f9_R1.fastq.gz f9_r2.fastq.gz

error: `f9_R1.fastq.gz': No such file or directory

make: *** [cgf9-1.fa] Error 1



Nnaemeka Emmanuel Nnadi,Ph.D
Department of Microbiology,
Faculty of Natural and Applied Science,
Plateau State University, Bokkos, Plateau State, Nigeria.
+2348068124819
Publications:
https://scholar.google.com/citations?user=5mzOgPQAAAAJ&hl=en
https://orcid.org/0000-0002-8424-4859

 https://www.researchgate.net/profile/Emmanuel_Nnadi/publications

**Comment by emmannaemeka** [comment]
Error still persists.

abyss-pe name=cgf9 k=96 B=2G in='/Users/emmanuelnnadi/Documents/Research/cglabrata/cglabarata_genome/a9/A9_R1.fastq.gz   
/Users/emmanuelnnadi/Documents/Research/cglabrata/cglabarata_genome/a9/A9_R2.fastq.gz'
AdjList    -k96 -m50 --dot cgf9-1.fa >cgf9-1.dot
cgf9-1.fa:0: warning: file is empty
abyss-rresolver-short  -b2G -f0.8 -j2 -k96 \
	 -h cgf9-1-rr --dot -c cgf9-1-rr.fa \
	-g cgf9-1-rr.dot cgf9-1.fa cgf9-1.dot /Users/emmanuelnnadi/Documents/Research/cglabrata/cglabarata_genome/a9/A9_R1.fastq.gz /Users/emmanuelnnadi/Documents/Research/cglabrata/cglabarata_genome/a9/A9_R2.fastq.gz 
Assertion failed: (num_vertices(g) > 0), function read_dot, file DotIO.h, line 307.
make: *** [cgf9-1-rr.dot] Abort trap: 6

Please how do I solve this?

Thank you
